server/src/main/java/io/druid/metadata/SQLMetadataConnector.java

```diff
@@ -141,6 +141,7 @@ public final boolean isTransientException(Throwable e)
     return e != null && (e instanceof SQLTransientException
                          || e instanceof SQLRecoverableException
                          || e instanceof UnableToObtainConnectionException
+                         || e instanceof UnableToExecuteStatementException
```

Comment:
@gianm this causes lots of retries for duplicate primary key entry errors that will never be transient unless someone manually cleans the DB